packages/utilities/psammead-test-helpers/index.test.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import * as testHelpers from '.';
+import * as testHelpersFromSrc from './src/index';
+
+const testHelpersExpectedExports = {
+  shouldMatchSnapshot: 'function',
+  shallowRender: 'function',
+  shouldShallowMatchSnapshot: 'function',
+  isNull: 'function',
+  testUtilityPackages: 'function',
+};
+
+const expectedExports = { testHelpers: testHelpersExpectedExports };
+
+const acutalExports = { testHelpers };
+
+const acutalExportsFromSrc = { testHelpers: testHelpersFromSrc };
+
+const testTheTestHelperErrorCases = testHelperMethod => {
+  const consoleLogSpy = jest.spyOn(global.console, 'log');
+
+  const actual = { utility: { foo: {} } };
+  const expected = { utility: { bar: {} } };
+
+  expect(() => {
+    testHelperMethod(actual, expected, 'testing');
+  }).toThrowError();
+  expect(consoleLogSpy).toHaveBeenCalledWith(
+    'No expected export value declared in unit tests for testing/utility.js - foo',
+  );
+
+  consoleLogSpy.mockRestore();
+};
+
+describe('Psammead test helpers', () => {
+  it('should test all the utility exports exist and are the correct type', () => {
+    testHelpers.testUtilityPackages(
+      acutalExports,
+      expectedExports,
+      'psammead-test-helpers',
+    );
+  });
+
+  it('should test all the utility exports exist and are the correct type when coming from src/', () => {
+    testHelpers.testUtilityPackages(
+      acutalExportsFromSrc,
+      expectedExports,
+      'psammead-test-helpers',
+    );
+  });
+
+  it('should error and console.log if expectedExport is missing a value compared to the actual export', () => {
+    testTheTestHelperErrorCases(testHelpers.testUtilityPackages);
+  });
+
+  it('should error and console.log if expectedExport is missing a value compared to the actual export when coming from /src', () => {
+    testTheTestHelperErrorCases(testHelpersFromSrc.testUtilityPackages);
+  });
+
+  expect(() => {
+    testHelpersFromSrc.isNull(
+      'test isNull method does not error when passed null',
+      null,
+    );
+  }).not.toThrowError();
+
+  expect(() => {
+    testHelpersFromSrc.isNull(
+      'test isNull method errors when passed a react component',
+      <p>foobar</p>,
+    );
+  }).toThrowError();
```

Comment:
This is a confusing problem, I'm trying to `expect()` that a method which has it's own `expect()` correctly fails. However, the method I want to test (which contains an `expect()`) fails before the wrapping `expect()` runs...

![image](https://user-images.githubusercontent.com/7791726/50108005-15dafb00-022c-11e9-9671-9810facd0946.png)

I was thinking about mocking the expect but surely that will cause the test to never run becasue `expect` will then be a mock?!

🤔🤔🤔🤔🤔🤔🤔

Reply:
We shouldn't be testing this IMO. If `foo.isNull()` is part of the test library API, we should trust it does what it says without having to verify it ourselves.

Reply:
Woohoo! I did think as much but I was hoping to get 100% test coverage 🥇